src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/KeyValuePairConverter.cs

```diff
@@ -107,14 +131,28 @@ internal override bool OnTryWrite(Utf8JsonWriter writer, KeyValuePair<TKey, TVal
         {
             writer.WriteStartObject();
 
-            writer.WritePropertyName(_keyName);
-            JsonSerializer.Serialize(writer, value.Key, options, ref state, KeyName);
+            writer.WritePropertyName(_keyNameEncoded);
+            JsonSerializer.Serialize(writer, value.Key, options, ref state, _keyName);
 
-            writer.WritePropertyName(_valueName);
-            JsonSerializer.Serialize(writer, value.Value, options, ref state, ValueName);
+            writer.WritePropertyName(_valueNameEncoded);
+            JsonSerializer.Serialize(writer, value.Value, options, ref state, _valueName);
 
             writer.WriteEndObject();
             return true;
         }
+
+        private bool FoundKeyProperty(string propertyName, bool caseInsensitiveMatch)
+        {
+            return propertyName == _keyName ||
+                (caseInsensitiveMatch && string.Equals(propertyName, _keyName, StringComparison.OrdinalIgnoreCase)) ||
+                propertyName == KeyNameCLR;
```

Comment:
We compare against `KeyNameCLR` and `ValueNameCLR` just to make this less of a breaking change, right? And hence, we should not honor `PropertyNameCaseInsensitive` similar to `_keyName` in this case, is that correct?

Reply:
Yes. `PropertyNameCaseInsensitive` applies to matching against the result of the conversion by the naming policy. We previously serialized only "Key" & "Value", so we only care about those two literals.